src/test/scala/CoreServerTests.scala

```diff
@@ -64,6 +64,10 @@ class CoreServerTest extends WordSpec with Matchers with ScalatestRouteTest {
 
   private val empty_args: Array[String] = Array()
 
+  // create a new execution context for each ViperCoreServer instance
+  // (otherwise there is a not-unique exception occurring for the Terminator actor when concurrently verifying multiple programs
+  private implicit def getExecutionContext: VerificationExecutionContext = new DefaultVerificationExecutionContext()
```

Comment:
@aterga do you might have an idea why reusing the same execution context / actor system does not work (i.e. when changing this function definition to a `val`?

Reply:
The comment above was added by Valentin, I think. Perhaps it's some simple issue, like two instances of the Terminator actor would occur in the same system with the same name? If that's the case, one could run multiple servers inside one actor system by making sure that we add some kind of unique server identifier as part of the name of all actors that we create. (Two actors wit the same name cannot exist in one actor system.)

Reply:
I would just try running any see what would fail.